services/apps/data_sink_worker/src/service/member.service.ts

```diff
@@ -392,6 +393,74 @@ export default class MemberService extends LoggerBase {
     }
   }
 
+  public async processMemberUpdate(
+    tenantId: string,
+    integrationId: string,
+    platform: PlatformType,
+    member: IMemberData,
+  ): Promise<void> {
+    this.log = getChildLogger('MemberService.processMemberUpdate', this.log, {
+      integrationId,
+      tenantId,
+    })
+
+    try {
+      this.log.debug('Processing member update.')
+
+      if (!member.identities || member.identities.length === 0) {
+        const errorMessage = `Member can't be updated. It is missing identities fields.`
+        this.log.warn(errorMessage)
+        return
+      }
+
+      await this.store.transactionally(async (txStore) => {
+        const txRepo = new MemberRepository(txStore, this.log)
+        const txIntegrationRepo = new IntegrationRepository(txStore, this.log)
+        const txService = new MemberService(
+          txStore,
+          this.nodejsWorkerEmitter,
+          this.searchSyncWorkerEmitter,
+          this.log,
+        )
+
+        const dbIntegration = await txIntegrationRepo.findById(integrationId)
+        const segmentId = dbIntegration.segmentId
+
+        // first try finding the member using the identity
+        const identity = singleOrDefault(member.identities, (i) => i.platform === platform)
```

Comment:
this'll throw an error if the member has multiple twitter identities - u can also filter by integrationId to filter the results more

Reply:
How do I filter by integrationId? I don't understand. Here we are filter identities coming from the integration, not from db. So there is only integrationId for all of them (if many)

In general, for Twitter this will never happen, so I guess it's fine.

Reply:
yeah true - for some reason i thought these were identities we got from db, resolving conversation